### PR TITLE
Replaces travis with GH actions for testing

### PR DESCRIPTION
**Pull request by cgranleese-r7**
Tests can be found [here](https://github.com/cgranleese-r7/metasploit-erd/actions/runs/789411165)

image

### CODE REVIEW

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -4,15 +4,10 @@
 require 'simplecov'
 require 'coveralls'
 
-# if ENV['TRAVIS'] == 'true'
-#   # don't generate local report as it is inaccessible on travis-ci, which is why coveralls is being used.
-#   SimpleCov.formatter = Coveralls::SimpleCov::Formatter
-# else
   SimpleCov.formatter = SimpleCov::Formatter::MultiFormatter.new(
```
Comment (adfoster-r7) [inline spec/spec_helper.rb:7]: Does this provide the same functionality?

```suggestion
if ENV['CI'] == 'true'
  # don't generate local report as it is inaccessible on travis-ci, which is why coveralls is being used.
  SimpleCov.formatter = Coveralls::SimpleCov::Formatter
else
  SimpleCov.formatter = SimpleCov::Formatter::MultiFormatter.new(
```

https://github.blog/changelog/2020-04-15-github-actions-sets-the-ci-environment-variable-to-true/

Comment (adfoster-r7) [reply spec/spec_helper.rb:7]: Or rather, what did this old code do, and do we need to make any changes here?

Comment (cgranleese-r7) [reply spec/spec_helper.rb:7]: It was originally commented out, so I just removed it as it was referring to Travis.

File: spec/spec_helper.rb [hunk 2]
```diff
@@ -4,15 +4,10 @@
...
       # either generate the local report
       SimpleCov::Formatter::HTMLFormatter
   )
```
Comment (adfoster-r7) [inline spec/spec_helper.rb:10]: We should dedent this 👀

File: spec/spec_helper.rb (hunk 1 above)
Comment (adfoster-r7) [reply spec/spec_helper.rb:7]: Can we delete the `coveralls` references elsewhere now? :+1:
Such as the gem requires and above in the spec helper

File: spec/spec_helper.rb (hunk 2 above)
Comment (cgranleese-r7) [reply spec/spec_helper.rb]: Got that sorted

File: .github/workflows/verify.yml [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+            ${{ runner.os }}-gems-
+      - name: Bundle install
+        run: |
+          bundle config path vendor/bundle
+          bundle install --jobs 4 --retry 3
+      - name: ${{ matrix.test_cmd }}
+        run: |
+          echo "${CMD}"
+          bash -c "${CMD}"
+
```
Comment (adfoster-r7) [inline .github/workflows/verify.yml:56]: This isn't needed now I don't think?

Comment (adfoster-r7) [reply .github/workflows/verify.yml:56]: The matrix doesn't define test_cmd, so this will be empty? 👀

Comment (cgranleese-r7) [reply .github/workflows/verify.yml]: Got it removed 👍

**Pull request merged**